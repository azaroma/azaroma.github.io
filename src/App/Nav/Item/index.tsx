import * as React from "react";
import { Link } from "react-router-dom";

export interface INavItem {
  name: string;
  route: string;
}

const NavElement: React.SFC<INavItem> = ({ name, route }) => (
  <li className="pa1 f4">
    <Link to={route} className="pa1 link black bg-animate hover-bg-gold">
      {name}
    </Link>
  </li>
);

export default NavElement;
