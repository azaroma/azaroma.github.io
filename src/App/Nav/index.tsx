import classnames from "classnames";
import * as React from "react";
import Fox from "../../assets/Fox.gif";

const Nav: React.SFC = () => {
  const classes = classnames(
    "bb",
    "pa3",
    "flex",
    "flex-column-reverse",
    "items-center",
    "flex-row-ns"
  );
  return (
    <nav className={classes}>
      <img className={"w2 w3-ns"} src={Fox} />
    </nav>
  );
};

export default Nav;
