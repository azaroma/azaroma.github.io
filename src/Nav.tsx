import * as React from "react";
import Fox from "./assets/Fox.gif";

const Nav: React.SFC = () => (
  <nav className={"pa3 bb"}>
    <img className={"w2 w3-ns"} src={Fox} />
  </nav>
);

export default Nav;
