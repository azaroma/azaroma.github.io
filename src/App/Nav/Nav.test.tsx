import { shallow } from "enzyme";
import * as React from "react";
import Nav from ".";

describe("Nav component", () => {
  const wrapper = shallow(<Nav />);

  it("renders correctly", () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toHaveLength(1);
  });

  it("looks as expected", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
