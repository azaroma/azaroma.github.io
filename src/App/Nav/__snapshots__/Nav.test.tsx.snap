// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component looks as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        className="w2 w3-ns"
        src={undefined}
      />,
      "className": "bb pa3 flex flex-column-reverse items-center flex-row-ns",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "w2 w3-ns",
        "src": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          className="w2 w3-ns"
          src={undefined}
        />,
        "className": "bb pa3 flex flex-column-reverse items-center flex-row-ns",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "w2 w3-ns",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
